src/effects.cpp

```diff
@@ -258,8 +258,8 @@ std::shared_ptr<LEDStripEffect> GetSpectrumAnalyzer(CRGB color)
 #define STARRYNIGHT_PROBABILITY 1.0
 #define STARRYNIGHT_MUSICFACTOR 1.0
 
-DRAM_ATTR std::unique_ptr<EffectFactories> g_ptrEffectFactories = nullptr;
-std::map<int, JSONEffectFactory> g_JsonStarryNightEffectFactories =
+static DRAM_ATTR std::unique_ptr<EffectFactories> l_ptrEffectFactories = nullptr;

```

Comment:
Prefix of l is weird!  We always uses s_ but I suppose some might confuse that with string, so I'll live with it :-)

Reply:
Maybe I can redeem myself a little bit by saying hand-on-heart that I did consider s_ as well. :)

Anyway, `static` still is the wrong keyword for a compilation unit variable.

Reply:
Wait, what?  Am I thinking K&R or what?  I was sure static meant "at global scope but accessible only here".